### ISSUE TITLE
Clean up copyright notices

### ISSUE THREAD

**Issue by drwells** [IBAMR/IBAMR#604]
We mark authorship and years somewhat inconsistently in headers. We should add a script that cleans up the headers and puts in the correct years.

Furthermore, though I am not an expert on the subject, the more common route for attribution is to have the header text simply state 'copyright (C) 2000 - 2019 by the IBAMR authors' and not list individual contributors. I think explicitly assigning ourselves copyright can avoid some legal problems. I am not sure how much of this UNC owns too.

**Comment by boyceg** [comment]
> We mark authorship and years somewhat inconsistently in headers. We should add a script that cleans up the headers and puts in the correct years.

Yes.

> Furthermore, though I am not an expert on the subject, the more common route for attribution is to have the header text simply state 'copyright (C) 2000 - 2019 by the IBAMR authors' and not list individual contributors.

Yes.

> I am not sure how much of this UNC owns too.

That is a question that I would rather not delve into. :-D

**Comment by drwells** [comment]
At other universities I have had to fill out intellectual property forms yearly. I won't tell if you don't :speak_no_evil:

**Comment by drwells** [comment]
On a related note: we should include a LICENSE file, which we do not currently have. I am *pretty sure* that we stated in a funding source that we should release under the LGPL. Is this correct?

**Comment by boyceg** [comment]
I think we use the [3-clause BSD license](https://opensource.org/licenses/BSD-3-Clause) --- see the top of any source code file in the library.

**Comment by drwells** [comment]
I thought that looked familiar. Thanks.

**Comment by drwells** [comment]
This was partially addressed by #646.

I spent some time this morning reading up on our copyright situation. AFAICT IBAMR is a non-directed work and therefore UNC does not make any claim to copyright: see

https://unc.policystat.com/policy/6197184/latest/
and
https://policy.uncg.edu/university-policies/copyright/
the second one is not UNC-CH, but our policy does not explicitly define some terms. I assume that there are no significant differences in IP management between different UNCs.

It is not clear to me if NYU claims copyright over older versions of IBAMR: see
https://www.nyu.edu/content/dam/nyu/compliance/documents/IPPolicy.pdf
see page 18 (section XI, subsection A1) and page 19 (section XI, subsection B1). I am pretty sure that the answer is 'no' but it depends on how @boyceg and others were funded at that time (and upon OIL, NYU's relevant legal team).

I suggest that we simply attribute copyright to "Boyce Griffith and the IBAMR authors": simply writing "The IBAMR authors" would also work.

See also:
https://opensource.stackexchange.com/questions/5508/what-does-and-contributors-in-the-copyright-byline-imply

To fix this problem more permanently we should require explicity copyright attribution from authors. The best way (in my opinion) to accomplish this is to require acknowledgement of the following for all contributions:
https://elinux.org/Developer_Certificate_Of_Origin

In conclusion, here is the roadmap for finishing this:
- [ ] Verify that neither NYU nor UNC assert copyright over parts of IBAMR. @boyceg does UNC make you sign any statements about the generation of intellectual property at UNC?
- [ ] Clarify what is meant by "The IBAMR authors". I will figure this out.
- [ ] Check that all contributors either assigned copyright to @boyceg (or another active developer) or are willing to assign copyright now to "The IBAMR authors".
- [ ] Permanently solve the problem by adding the Linux Foundation's Developer Certificate of Origin.

**Comment by drwells** [comment]
AFAICT "The IBAMR authors" will legally refer to anyone who has made fundamental or substantial contributions to the library. See, e.g., the discussion in
https://google.github.io/opencasebook/authorship/
(where I assume IBAMR is a joint work).

**Comment by boyceg** [comment]
You are going to make me talk to lawyers, aren't you...

**Comment by drwells** [comment]
Hopefully I can handle this!

**Comment by drwells** [comment]
I thought about this some more and came to the following conclusion: copyright statements in code are not required for copyright to exist. If NYU wants to claim copyright on older versions of IBAMR they can do so without needing us to mention them in the preamble.

AFAICT neither UNC's or NYU's policies indicate that either one of them have any intent of making such a claim anyway.

Hence, in conclusion: they can claim copyright without the preamble anyway and are extremely unlikely to do so: I think that we shouldn't bother OIL or OTC about the status of IBAMR.

**Comment by drwells** [comment]
@boyceg Yes or no: shall we replace the copyright notice with
```cpp
// Copyright (c) 2002-2017, the IBAMR developers
// All rights reserved.
//
// This file is part of IBAMR.
//
// IBAMR is free software and distributed under the 3-clause BSD license.
// The full text of the license can be found in the file LICENSE.md at
// the top level directory of IBAMR.
```
We can use the version control information to get the year information.

**Comment by boyceg** [comment]
Sounds good to me.

**Comment by boyceg** [comment]
Although shall we make it 2019 instead of 2017? :-)

**Comment by drwells** [comment]
Well, yes, but we can ask git to print the first and last years that a file was modified.

**Comment by boyceg** [comment]
Sure, as long as it is easy to automate. Should there be a CD job that keeps this stuff up-to-date?

**Comment by drwells** [comment]
We have a script in deal.II that we run before each release that fixes the years. I don't think its worth worrying about more often than that.
